### DISCUSSION TITLE
unable to use Vercel for deployment properly

### DISCUSSION THREAD

**Discussion by gusibi** [gethinode/hinode#789] [Help] [answered]
**Problem Description**

I'm using the Hinode template and encountering an issue when deploying it on Vercel. The generated site is in XML format instead of HTML format. I would like the site to be correctly rendered in HTML format. I have checked the configuration files and relevant documentation for the Hinode template but haven't been able to resolve the issue.

**Steps to Reproduce**

1. Generate the site locally using the Hinode template.
2. Deploy the site to Vercel.
3. Access the deployed site on Vercel and observe that it is in XML format instead of HTML format.

**Expected Result**

I expect the site deployed on Vercel to generate pages in HTML format.

**Actual Result**

The generated site is in XML format instead of HTML format.

**Attempted Solutions**

- Checked the configuration files of the Hinode template to ensure no misconfiguration.
- Consulted the documentation of the Hinode template for any relevant information.
- Attempted some changes in the Vercel configuration, but without success.

**Additional Information**

https://github.com/gusibi/sora-vedio

 log outputs, or code snippets.

   Building sites … WARN 2024/02/21 01:44:57 found no layout file for "HTML" for "page": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.

**Comment by markdumay** [comment]
I checked your repository, but it looks like your site is misconfigured. I cannot start a local server, as Hugo enters in an infinite loop when downloading the modules.

I'd recommend to follow the [Hinode installation instructions.](https://gethinode.com/docs/getting-started/introduction/#installation) If you're not into automation, you can follow the `Hugo` instead of the `npm` instructions.

**Reply by gusibi** [reply]
thanks for your reply.

I configured it using the recommended method on the dev branch, but the deployment was unsuccessful.
https://github.com/gusibi/sora-vedio/tree/dev
I then tested it using the solution recommended by Hugo's official documentation, but it still didn't work.

**Reply by markdumay** [reply]
I've made a few adjustments to your base config. I can run the site locally now. Would you mind sharing a link to the Vercel deployment steps? I can't find any instructions on gohugio.io, which is the official Hugo site.

**Reply by gusibi** [reply]
I apologize for the confusion. 
I have used the official documentation of Hugo to create a main branch for deployment on Vercel.

https://gohugo.io/getting-started/quick-start/

I am using the default configuration of Vercel, but I am receiving a prompt stating that there is no HTML. 
Below is a screenshot of my configuration.![image](https://github.com/gethinode/hinode/assets/1282277/3238b36f-4232-456f-93f9-471a3b0a8eaf)

![image](https://github.com/gethinode/hinode/assets/1282277/7487b0f6-d868-44b2-ae3f-12f3a46c2bf6)

**Reply by gusibi** [reply, marked as answer]
I have found the issue, it is related to the version of Hugo on Vercel. After specifying the version, the deployment was successful. 

Thank you for your assistance. 
I am very grateful.

![image](https://github.com/gethinode/hinode/assets/1282277/af1e1303-5125-42a9-b087-a1f930b34f2b)

**Reply by markdumay** [reply]
Great! I did some testing too, and indeed setting `HUGO_VERSION` to `0.122.0` (or later) does the trick. As an additional suggestion, some of the Hinode functions require the modules to be vendored. One such example is the [purging of unused stylesheets elements](https://gethinode.com/guides/optimization/). This feature requires npm, however, you might want to include the following settings to your Vercel configuration anyways.

![vercel](https://github.com/gethinode/hinode/assets/61946753/30f1771f-488b-401d-9b6f-2ae70e7d9734)

You can stack multiple commands using the `&&` syntax. The command `hugo mod vendor` requires Go to be installed on the Vercel runner (it is not by default). You can use `yum install golang` as install command to do so.

**Reply by gusibi** [reply]
Thank you, I added the configuration and now everything is working perfectly. 
By the way, if I want to add some custom HTML to the page, how can I do that?

**Reply by markdumay** [reply]
By default, HTML support is disabled. You can enable this by adjusting the [Goldmark configuration](https://gethinode.com/docs/content/content-management/#mixed-content). You can then simply add HTML code to your markdown file.

**Reply by gusibi** [reply]
Thanks a lot. Have a nice day

**Comment by markdumay** [comment]
As an update in July 2025, make sure to add the `-y` flag to the install command, e.g. `yum install -y golang`. This prevents a build timeout (without any build errors).
